### PR TITLE
Enh/parallel analysis

### PR DESCRIPTION
**Pull request by nathlacroix**
Introduces an AnalysisDaemon to produce plots for analysis without interrupting measurements. Thanks to @chellings for the draft implementation. The following steps are required to run the daemon for basic calibration measurements:
1. create an analysis daemon, either by instanciating a class `AnalysisDaemon()` in a separate jupyter notebook (with the pycqed environment, and the correct datadirectory set in `a_tools`) or through commandline, again with appropritate pycqed environment started. e.g. `python analysis_daemon.py --data-dir D:\pydata` . A starting timestamp for the watching of the datafolder can be optionally specified witht the flag `--t-start`. Otherwise, the daemon will start watching any folder created after it the daemon was started. In `pycqedscripts/scripts`, there is a script called `start_analysis_daemon.bat` that starts the daemon with default settings for XLD/BF1. 
2. Pass the flag `delegate_plotting=True` through the options_dict to the analysis. This pull request adds the "delegate_plotting" flag to all basic calibration measurements functions in mqm, i.e. mutiqubit rabi, ramsey, qscale, t1, t2echo, ssro. 

Inviting @chellings to review. FYI @christiankraglundandersen @antsr @stephlazar

### CODE REVIEW

File: pycqed/analysis_v2/analysis_daemon.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                                           return_path=False,
+                                           return_timestamp=True, n_matches=1)[0]
+        self.run()
+
+    def run(self):
+        try:
+            while (True):
+                self.check_job()
+                for i in range(self.poll_interval):
+                    time.sleep(1)
```
Comment (antsr) [inline pycqed/analysis_v2/analysis_daemon.py:65]: why not just `time.sleep(self.poll_interval)`?

**Review by antsr — COMMENTED** [review]
Very nice addition!

File: pycqed/analysis_v2/analysis_daemon.py (hunk 1 above)
Comment (chellings) [reply pycqed/analysis_v2/analysis_daemon.py:65]: because you can KeyboardInterrupt only at the end of a sleep

Comment (nathlacroix) [reply pycqed/analysis_v2/analysis_daemon.py:65]: Yes as Christoph points out, time.sleep() does not release the lock and therefore exceptions cannot be raised during the sleeping

**Review by chellings — APPROVED** [review]
Looks good to me and will be a useful feature. Thanks @nathlacroix for turning my hacky draft into useful code! I already had the chance to comment directly during the development process, so I have no significant further comments here.

Maybe just one thing: if someone want to call the AnalysisDaemon from a notebook, a docstring for the class might be helpful (for starting from command line the explanations in add_argument should be sufficient).

**Comment by nathlacroix** [comment]
@antsr see point 2. here for which measurements already support the delegate_plotting from the "highest" pycqed layer. 

In addition, any child class of  MultiQubitTimeDomainAnalysis can currently support the delegate_plotting in the options_dict, as long as this child class does not have (keyword) arguments indispensible to its successful run which are not passed down to the parent mqtda.__init__(). i.e. : 
```
class Child(MultiQubitTimeDomainAnalysis):
    def __init__(self, child_kwarg="", **kwargs):
        super().__init__(**kwargs)
        # (...) something with child_kwarg
```
==> child_kwarg won't be included in the job. 

Classes derived from BaseDataAnalysis can easily create their own job (just like MultiQubitTimeDomainAnalysis) by calling create_job(...) and passing all args and kwargs at the end of their init.

**Pull request merged**